lib/chef/recipe.rb

```diff
@@ -88,26 +89,24 @@ def untag(*tags)
     def from_yaml_file(filename)
       self.source_file = filename
       if File.file?(filename) && File.readable?(filename)
-        from_yaml(IO.read(filename))
+        yaml_contents = IO.read(filename)
+        if ::YAML.load_stream(yaml_contents).length > 1
+          raise ArgumentError, "YAML recipe '#{filename}' contains multiple documents, only one is supported"
+        end
```

Comment:
TIL, I had no idea this was a thing.

Reply:
YIL YAML is based off rfc822 (ARPA INTERNET TEXT MESSAGES 1982).